src/git.rs

```diff
@@ -300,6 +375,58 @@ impl Git for LiveGit {
     }
 }
 
+impl LiveGit {
+    fn is_identical_tree_to_any_parent(commit: &Commit<'_>) -> bool {
+        let tree_id = commit.tree_id();
+        commit.parents().any(|p| p.tree_id() == tree_id)
+    }
+
+    fn get_tags(&self, commit_id: Oid, filter: &Option<String>) -> Vec<Tag> {
+        let mut tag_cache = self.tag_cache.borrow_mut();
+
+        tag_cache
+            .entry(filter.clone())
+            .or_insert_with(|| {
+                let all_tag_names = self.repo.tag_names(filter.as_deref()).ok();
+                let libgit_tags = all_tag_names
+                    .map(|tag_names| {
+                        tag_names
+                            .iter()
+                            .filter_map(|maybe_tag_name| maybe_tag_name)
```

Comment:
Is there an identity function?

Reply:
Yes! https://doc.rust-lang.org/std/convert/fn.identity.html
(replying to this one with the link because that isn't what I actually did in this one particular case)